### PR TITLE
Add support for cached, delegated mounts

### PR DESCRIPTION
**Pull request by mheon**
And by "support" I mean that we ignore them.

These are OS X only options from a recent Docker that are ignored on other OSes. Because it's safe everywhere, people are starting to use them, so let's add support for ignoring them, as Docker does.

Also includes some small cleanups in how we handle mount options.

### CODE REVIEW

File: pkg/spec/storage.go [hunk 1]
```diff
@@ -575,35 +579,52 @@ func ValidateVolumeCtrDir(ctrDir string) error {
...
 	var foundRootPropagation, foundRWRO, foundLabelChange, bindType int
+	finalOpts := make([]string, 0, len(options))
+	discardOpts := []string{"cached", "delegated"}
 	for _, opt := range options {
+		// The discarded ops are OS X specific volume options introduced
+		// in a recent Docker version.
+		// They have no meaning on Linux, so here we silently drop them.
+		// This matches Docker's behavior (the options are intended to
+		// be always safe to use, even not on OS X).
+		bad := false
```
Comment (vrothberg) [inline pkg/spec/storage.go]: Nit: `s/bad/ignore/`

But somehow, I'd prefer to have that in the switch table below to let the compiler properly optimize.
```golang
case "cached", "delegated":
    // Comment ...
    continue
```

Comment (rhatdan) [reply pkg/spec/storage.go]: I agree this seems like a simpler solution.

Comment (rhatdan) [reply pkg/spec/storage.go]: I agree this seems like a simpler solution.

@QiWang19 @ashley-cui FYI

Comment (mheon) [reply pkg/spec/storage.go]: Good catch. Pushed an updated version.

**Comment by rhatdan** [comment]
/lgtm
Now we need @ashley-cui and @jwhonce to figure out what these do on a MAC.

**Pull request merged**